### PR TITLE
[BEAM-7389] Add code examples for Filter page

### PR DESCRIPTION
**Pull request by davidcavazos**
Adding code examples for Filter page.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by davidcavazos** [comment]
Staged: http://apache-beam-website-pull-requests.storage.googleapis.com/9184/documentation/transforms/python/elementwise/filter/index.html

**Comment by davidcavazos** [comment]
R: @rosetn can you take a look whenever you have a chance?

**Comment by aaltay** [comment]
Please ping me after @rosetn reviews.

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 1]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+  </td>
+</table>
+<br>
+
+> **Note**: It is also possible to pass the `PCollection` as a *list* with `beam.pvalue.AsList(pcollection)`,
+> but this will require all the elements to fit into memory.
+
+### Side Inputs - Dict
+
+If the `PCollection` consists of a small enough number of key-value pairs that can fit into memory,
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/filter.md]: I recommend simplifying if this sentence is still accurate:

"If a `PCollection` is small enough to fit into memory, then that `PCollection` can be passed as a *dictionary*."

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 2]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
 May also be used to filter based on an inequality with a given value based
 on the comparison ordering of the element.
 
 ## Examples
-See [BEAM-7389](https://issues.apache.org/jira/browse/BEAM-7389) for updates. 
 
-## Related transforms  
+### Function
+
+`Filter` accepts a function that receives the element as the first argument,
```
Comment (aaltay) [inline website/src/documentation/transforms/python/element-wise/filter.md]: `as the first argument` -> `as its first argument` ?

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 3]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+    </a>
+  </td>
+</table>
+<br>
+
+### Side Inputs - Iterator
+
+If there are multiple values in the `PCollection`, it is recommended way to pass the `PCollection` as an *iterator*.
+This will access elements lazily as they are needed,
+so it is possible to iterate over very large `PCollection`s that don't fit into memory.
```
Comment (aaltay) [inline website/src/documentation/transforms/python/element-wise/filter.md]: `don't fit into memory.` -> `won't fit into memory.` ?

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 4]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+
+> **Note**: It is also possible to pass the `PCollection` as a *list* with `beam.pvalue.AsList(pcollection)`,
+> but this will require all the elements to fit into memory.
+
+### Side Inputs - Dict
+
+If the `PCollection` consists of a small enough number of key-value pairs that can fit into memory,
+the `PCollection` can be passed as a *dictionary*.
+Note that all the elements of the `PCollection` must fit into memory for this.
+If the `PCollection` can be very large, use `beam.pvalue.AsIter(pcollection)` instead.
```
Comment (aaltay) [inline website/src/documentation/transforms/python/element-wise/filter.md]: `can be very large,` -> `won't fit in memory,` ? (to use the similar terminology as above.)

File: website/src/documentation/transforms/python/element-wise/filter.md (hunk 2 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Ended up rewriting it closer to what Rose suggested.

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 5]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+        href="https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py">
+      <img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png"
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Side Inputs - Singleton
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 6]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+        href="https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py">
+      <img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png"
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Side Inputs - Iterator
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 7]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+
+If there are multiple values in the `PCollection`, it is recommended way to pass the `PCollection` as an *iterator*.
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 8]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+This will access elements lazily as they are needed,
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Makes sense, thank you!

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 9]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+        href="https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/transforms/element_wise/filter.py">
+      <img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png"
+        width="20px" height="20px" alt="View on GitHub" />
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+> **Note**: It is also possible to pass the `PCollection` as a *list* with `beam.pvalue.AsList(pcollection)`,
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 10]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+> but this will require all the elements to fit into memory.
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md [hunk 11]
```diff
@@ -18,25 +18,168 @@ See the License for the specific language governing permissions and
...
+
+### Side Inputs - Dict
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Done

File: website/src/documentation/transforms/python/element-wise/filter.md (hunk 1 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Sounds good, changed

File: website/src/documentation/transforms/python/element-wise/filter.md (hunk 4 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/filter.md]: Makes sense, changed

**Comment by davidcavazos** [comment]
@aaltay All tests have passed :)

**Pull request merged**